src/main/iidxhook3/avs-boot.c

```diff
@@ -0,0 +1,178 @@
+#define LOG_MODULE "avs-boot"
+
+#include <stdint.h>
+#include <string.h>
+
+#include "hook/table.h"
+
+#include "imports/avs.h"
+
+#include "iidxhook3/avs-boot.h"
+
+#include "util/log.h"
+
+static void (*real_avs_boot)(
+    struct property_node *config,
+    void *std_heap,
+    size_t sz_std_heap,
+    void *avs_heap,
+    size_t sz_avs_heap,
+    avs_log_writer_t log_writer,
+    void *log_context);
+static int (*real_ea3_boot_avs)(struct property_node *config);
+static int (*real_ea3_boot)(struct property_node *config);
+
+static void my_avs_boot(
+    struct property_node *config,
+    void *std_heap,
+    size_t sz_std_heap,
+    void *avs_heap,
+    size_t sz_avs_heap,
+    avs_log_writer_t log_writer,
+    void *log_context);
+static int my_ea3_boot_avs(struct property_node *config);
+static int my_ea3_boot(struct property_node *config);
+
+static struct net_addr iidxhook3_avs_boot_eamuse_server_addr;
+
+static const struct hook_symbol iidxhook3_avs_hook_syms[] = {
+    {.name = "avs_boot",
+     .patch = my_avs_boot,
+     .link = (void **) &real_avs_boot},
+    {.name = "ea3_boot",
+     .patch = my_ea3_boot_avs,
+     .link = (void **) &real_ea3_boot_avs},
+};
+
+static const struct hook_symbol iidxhook3_ea3_hook_syms[] = {
+    {.name = "ea3_boot",
+     .patch = my_ea3_boot,
+     .link = (void **) &real_ea3_boot},
+};
+
+static void avs_boot_replace_property_uint32(
+    struct property_node *node, const char *name, uint32_t val)
+{
+    struct property_node *tmp;
+
+    tmp = property_search(NULL, node, name);
+
+    if (tmp) {
+        property_node_remove(tmp);
+    }
+
+    property_node_create(NULL, node, PSMAP_TYPE_U32, name, val);
+}
+
+static void avs_boot_replace_property_str(
+    struct property_node *node, const char *name, const char *val)
+{
+    struct property_node *tmp;
+
+    tmp = property_search(NULL, node, name);
+
+    if (tmp) {
+        property_node_remove(tmp);
+    }
+
+    tmp = property_node_create(NULL, node, PROPERTY_TYPE_STR, name, val);
+
+    if (tmp) {
+        property_node_datasize(tmp);
+    } else {
+        log_fatal("Could not avs_boot_replace_property_str(%s, %s)", name, val);
+    }
+}
+
+static void my_avs_boot(
+    struct property_node *config,
+    void *std_heap,
+    size_t sz_std_heap,
+    void *avs_heap,
+    size_t sz_avs_heap,
+    avs_log_writer_t log_writer,
+    void *log_context)
+{
+    log_info("Called my_avs_boot");
+
+    avs_boot_replace_property_uint32(config, "log/level", 4);
```

Comment:
suggestion: Can you adjust the log level to align with what's configured for the hook? That would ensure that the log level can be controlled properly, i.e. to reduce output on normal/production operation.

Reply:
What specific steps should be taken?
For example, making it configurable via conf file?

The reason for hardcoding the log level was to follow the existing implementation.

https://github.com/djhackersdev/bemanitools/blob/master/src/main/jbhook1/avs-boot.c#L118
https://github.com/djhackersdev/bemanitools/blob/master/src/main/iidxhook4-cn/avs-boot.c#L81

Reply:
I see, that's in general an issue that the current setup with logging does not allow for centralized log level management on the older games. I believe this requires more effort to be re-worked properly. Thus, I consider your approach valid and reasonable. Thanks for clarifying.